src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -141,12 +141,12 @@ SecurityHandshaker::~SecurityHandshaker() {
   gpr_mu_destroy(&mu_);
   tsi_handshaker_destroy(handshaker_);
   tsi_handshaker_result_destroy(handshaker_result_);
-  if (endpoint_to_destroy_ != nullptr) {
```

Comment:
IIC this is has some behavior change, since we're now unconditionally destroying `args_->endpoint` and `args_->read_buffer`, where before they would only be destroyed upon handshake failure?

Reply:
You are right. Fixed it